java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/registration/SchemaTopicBackup.java

```diff
@@ -0,0 +1,214 @@
+package org.radarcns.schema.registration;
+
+import static org.apache.kafka.common.config.TopicConfig.CLEANUP_POLICY_COMPACT;
+import static org.apache.kafka.common.config.TopicConfig.CLEANUP_POLICY_CONFIG;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.confluent.kafka.schemaregistry.storage.SchemaRegistryKey;
+import io.confluent.kafka.schemaregistry.storage.SchemaRegistryKeyType;
+import io.confluent.kafka.schemaregistry.storage.SchemaRegistryValue;
+import io.confluent.kafka.schemaregistry.storage.SchemaValue;
+import io.confluent.kafka.schemaregistry.storage.exceptions.SerializationException;
+import io.confluent.kafka.schemaregistry.storage.serialization.SchemaRegistrySerializer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.constraints.NotNull;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+/** Data class for containing the data and metadata of the _schemas topic. */
+public class SchemaTopicBackup {
+    @JsonProperty
+    private final Map<String, String> settings;
+    @JsonProperty
+    private final Set<SchemaRecord> records;
+
+    /** Empty backup. */
+    public SchemaTopicBackup() {
+        settings = new HashMap<>();
+        records = new LinkedHashSet<>();
+    }
+
+    /** Fully ready backup. */
+    @JsonCreator
+    public SchemaTopicBackup(
+            @JsonProperty("settings") Map<String, String> settings,
+            @JsonProperty("records") List<SchemaRecord> records) {
+        this.settings = new HashMap<>(settings);
+        this.records = new LinkedHashSet<>(records);
+    }
+
+    /**
+     * Whether the backup contains schemas starting from ID 1. If so, the schema topic is valid.
+     */
+    public boolean startsAtFirstId() {
+        return records.stream()
+                .map(SchemaRecord::getSchemaId)
+                .filter(Objects::nonNull)
+                .anyMatch(r -> r == 1);
+    }
+
+    @JsonGetter
+    @NotNull
+    public Map<String, String> getSettings() {
+        return settings;
+    }
+
+    /**
+     * Put settings.
+     *
+     * @param settings Kafka topic config settings
+     */
+    public void putAll(@NotNull Map<? extends String, ? extends String> settings) {
+        this.settings.putAll(settings);
+        if (!Objects.equals(settings.get(CLEANUP_POLICY_CONFIG), CLEANUP_POLICY_COMPACT)) {
```

Comment:
why do we force set this to `CLEANUP_POLICY_COMPACT`?

Reply:
Because if the cleanup policy is `DELETE`, data will be deleted after a set amount of time. With compact strategy, Kafka should not actively delete any data in the topic if it has a unique key.